extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -113,21 +128,44 @@ export class RegexValidation extends Validation {
 }
 
 export abstract class Comparison extends Validation {
-	protected _target: string; // comparison object require a target so override the base optional setting.
+	protected _target: string; // comparison object requires a target so override the base optional setting.
+	protected _onTargetValidityChangedGetter: OnTargetValidityChangedGetter; // comparison object requires an onTargetValidityChangedGetter defined so override the base optional setting.
+	protected _onNewDisposableCreated: (disposable: vscode.Disposable) => void; // comparison object requires an (disposable: vscode.Disposable) => void defined so override the base optional setting.
+	protected _listenerOnTargetValidityChangeAdded = false;
 
 	get target(): string {
 		return this._target;
 	}
 
-	constructor(validation: ComparisonValidationInfo, valueGetter: ValueGetter, variableValueGetter: VariableValueGetter) {
-		super(validation, valueGetter, variableValueGetter);
+	protected onTargetValidityChanged(onTargetValidityChangedAction: (e: boolean) => Promise<void>): void {
+		const onValidityChanged = this._onTargetValidityChangedGetter(this.target);
+		this._onNewDisposableCreated(onValidityChanged(isValid => onTargetValidityChangedAction(isValid)));
+	}
+
+	constructor(validation: ComparisonValidationInfo, validationMessageUpdater: onValidation, valueGetter: ValueGetter, targetValueGetter: TargetValueGetter, onTargetValidityChangedGetter: OnTargetValidityChangedGetter, onDisposableCreated: (disposable: vscode.Disposable) => void) {
+		super(validation, validationMessageUpdater, valueGetter, targetValueGetter);
+		this._onNewDisposableCreated = onDisposableCreated;
+		this._onTargetValidityChangedGetter = onTargetValidityChangedGetter;
 		throwUnless(validation.target !== undefined);
 		this._target = validation.target;
 	}
 
+	private validateOnTargetValidityChange() {
+		if (!this._listenerOnTargetValidityChangeAdded) {
```

Comment:
Why do you have this check? It seems bad to just silently make this call do nothing. Although I'm also not even why this is necessary - why wouldn't we hook into this right away instead of waiting until validate is called?

Reply:
By right away, I think you mean in the constructor of the validation object. It is not done there because at that time all fields may not yet have been populated/created, while the reference to (the variable Name that the field points to) is known. So, it cannot be done there. I will put in a comment in the constructor about it.

The check is there because we need to register the callback only once and not every time validation is processed. Right now, the 'validateOnTargetValidChange subscription is made of the first validity check instead of in the constructor for the reason mentioned above.

Reply:
Ok - not my favorite way of approaching it but it should be fine. I would suggest renaming this to `ensureOnTargetValidityChangeListenerAdded` though - the current name isn't very clear what it does.

Reply:
Agree on this not being the best way to do this. I will file a followup item for cleanup. Perhaps we can use the pendingActions paradigm here.

Reply:
changed name to _ensureOnTargetValidityChangeListenerAdded

Reply:
tracking issue to clean this up: #13456